src/h_vialib/client.py

```diff
@@ -0,0 +1,62 @@
+"""Helper classes for clients using Via proxying."""
+
+import re
+from urllib.parse import urlencode, urlparse
+
+
+class ViaDoc:  # pylint: disable=too-few-public-methods
+    """A doc we want to proxy with content type."""
+
+    GOOGLE_DRIVE_REGEX = re.compile(
+        r"^https://drive.google.com/uc\?id=(.*)&export=download$", re.IGNORECASE
+    )
+
+    def __init__(self, url, content_type=None):
+        """Initialize a new doc with it's url and content_type if known."""
+        self.url = url
+
+        if content_type is None and self.GOOGLE_DRIVE_REGEX.match(url):
+            content_type = "pdf"
+
+        self._content_type = content_type
+
+    @property
+    def is_pdf(self):
+        """Check if document is known to be a pdf."""
+        return self._content_type == "pdf"
```

Comment:
I'm linking enums more and more for this sort of thing, but this is a straight lift and shift so... fix if you want to, ignore if you don't!

Reply:
Yeah, fan of the Enums too but here I found `content_type` confusing, if I just read the function signature I'd have pass `application/pdf` so I just left as is for now